virttest/utils_kernel_module.py

```diff
@@ -0,0 +1,188 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2020
+# Author: Sebastian Mitterle <smitterl@redhat.com>
+
+"""
+Handle Linux kernel modules
+"""
+
+import os
+import logging
+
+from avocado.utils import process
+from avocado.core import exceptions
+
+
+def reload(module_name, force, params):
+    """
+    Convenience method that creates a KernelModuleHandler instance
+    and reloads the module only if any action will is required.
+
+    :param module_name: name of kernel module to be handled
+    :param force: if to force load with params in any case, e.g. True
+    :param params: parameters to load with, e.g. 'key1=param1 key2=param2 ...'
+    :return: instance if module was loaded
+    """
+
+    if params != "" or force:
+        handler = KernelModuleHandler(module_name)
+        handler.reload_module(force, params)
+        return handler
+    return None
+
+
+class KernelModuleHandler(object):
+    """Class handling Linux kernel modules"""
+
+    def __init__(self, module_name):
+        """Create kernel module handler"""
+
+        self._module_name = module_name
+        self._was_loaded = None
+        self._loaded_config = None
+        self._config_backup = None
+        self._backup_config()
+
+    def reload_module(self, force, params):
+        """
+        Reload module with given parameters.
+
+        If force=False loading will be skipped when either the module is already loaded
+        with the passed parameters or when the module has not been loaded at all.
+
+        +-----------------------+-+-+-+-+-+-+
+        |**precondition**                   |
+        +-----------------------+-+-+-+-+-+-+
+        |module loaded          |N|N|Y|Y|Y|Y|
+        +-----------------------+-+-+-+-+-+-+
+        |params already loaded  |*|*|Y|N|Y|N|
+        +-----------------------+-+-+-+-+-+-+
+        |force load             |Y|N|Y|Y|N|N|
+        +-----------------------+-+-+-+-+-+-+
+        |**result**                         |
+        +-----------------------+-+-+-+-+-+-+
+        |load module with params|Y|N|Y|Y|N|Y|
+        +-----------------------+-+-+-+-+-+-+
```

Comment:
How about more condense format:

```
        +-----------------------+-+-+-+-+-+-+
        |module loaded          |N|N|Y|Y|Y|Y|
        |params already loaded  |*|*|Y|N|Y|N|
        |force load             |Y|N|Y|Y|N|N|
        +-----------------------+-+-+-+-+-+-+
        |issue reload           |Y|N|Y|Y|N|Y|
        +-----------------------+-+-+-+-+-+-+
```
(also suggesting using "issue reload" instead of "load module with params" but that's just a suggestion)

Reply:
@ldoktor I'll change to `issue reload` (and `issue restore` for consistency); however, formatting the table that way will break the table when building doc through `cd doc && make html` (not only not nice to look at but really _broken_, believe me). So, I'll keep this format.